clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -6826,18 +6828,41 @@ class FreeFunctionPrinter {
       CTN.getAsTemplateDecl()->printQualifiedName(ParmListOstream);
       ParmListOstream << "<";
 
-      auto SpecArgs = TST->template_arguments();
-      auto DeclArgs = CTST->template_arguments();
+      ArrayRef<TemplateArgument> SpecArgs = TST->template_arguments();
+      ArrayRef<TemplateArgument> DeclArgs = CTST->template_arguments();
+
+      auto TemplateArgPrinter = [&](const TemplateArgument &Arg) {
+        if (Arg.getKind() != TemplateArgument::ArgKind::Expression ||
+            Arg.isInstantiationDependent()) {
+          Arg.print(Policy, ParmListOstream, false /* IncludeType */);
```

Comment:
I think this is the other way around
```suggestion
          Arg.print(Policy, ParmListOstream, /* IncludeType= */false);
```